docs/articles_en/documentation/openvino_extensibility/stateful_models_intro.rst

```diff
@@ -0,0 +1,106 @@
+ we can process it with RNN like models that contain a cycle inside. But in some cases, like online speech recognition of time series 
+ forecasting, length of data sequence is unknown. Then data can be divided in small portions and processed step-by-step. But dependency 
+ between data portions should be addressed. For that, models save some data between inferences - state. When one dependent sequence is over,
+ state should be reset to initial value and new sequence can be started.
+ 
+Deep learning frameworks provide a dedicated API to build models with state. For example, Keras has special option for RNNs `stateful` that turns on saving state 
+ between inferences. Kaldi contains special specifier `Offset` to define time offset in a model. 
+ 
+ OpenVINO also contains special API to simplify work with models with states. State is automatically saved between inferences, 
+ and there is a way to reset state when needed. You can also read state or set it to some new value between inferences.
+ 
+## OpenVINO State Representation
+
+ OpenVINO contains a special abstraction `Variable` to represent a state in a model. There are two operations to work with the state: 
+* `Assign` to save value to the state
+* `ReadValue` to read value saved on previous iteration
+
+![state_model_example](./img/state_model_example.png)
+
+The left side of the picture shows the usual inputs and outputs to the model: Parameter/Result operations. They have no connection with each other and in order to copy data from output to input users need to put extra effort writing and maintaining additional code.
+In addition, this may impose additional overhead due to data representation conversion.
+
+Having operations such as ReadValue and Assign allows users to replace the looped Parameter/Result pairs of operations and shift the work of copying data to OpenVINO. After the replacement, the OpenVINO model no longer contains inputs and outputs with such names, all internal work on data copying is hidden from the user, but data from the intermediate inference can always be retrieved using State API methods.
+
+In some cases, users need to set an initial value for State, or it may be necessary to reset the value of State at a certain inference to the initial value. For such situations, an initializing subgraph for the ReadValue operation and a special "reset" method are provided.
+
+You can find more details on these operations in [ReadValue specification](../ops/infrastructure/ReadValue_3.md) and 
+[Assign specification](../ops/infrastructure/Assign_3.md).
+
+## How to get the OpenVINO Model with states
+
+* [Convert Kaldi model to IR via Model Optimizer.](../MO_DG/prepare_model/convert_model/kaldi_specific)
+   If the original Kaldi model contains RNN-like operations with `stateful` option, then after ModelOptimizer conversion,
+   the resulting OpenVINO model will also contain states.
```

Comment:
Do not use this example. Remove all references to Kaldi. Instead, we will use optimum-intel as the main example of source for stateful models. I will cover that part, but I please remove this deprecated stuff from the update document.

Reply:
deleted